lib/local/environment_report.F90

```diff
@@ -126,73 +80,13 @@ end function getcwd_c
-        write (io_unit,'(1X,"git sha1 hash:",/,5X,a)') HANDE_VCS_VERSION
-
-        stat = gethostname(str, str_len)
-        if (stat == 0) then
-            write (io_unit, '(a10)') 'Hostname:'
-        else
-            write (io_unit, '(a53)') 'Hostname exceeds 255 characters; truncated hostname:'
-        end if
-        write (io_unit,'(5X,a)') trim(carray_to_fstring(str))
-
-        path = getcwd_c(str, str_len)
-        if (c_associated(path)) then
-            write (io_unit,'(a20)') 'Working directory: '
-        else
-            write (io_unit,'(a63)') 'Working directory exceeds 255 characters; truncated directory:'
-        end if
-        write (io_unit,'(5X,a)') trim(carray_to_fstring(str))
+        call print_info
 
         call date_and_time(VALUES=date_values)
 
-        write (io_unit,'(1X,a18,1X,i2.2,"/",i2.2,"/",i4.4,1X,a2,1X,i2.2,2(":",i2.2))') &
+        write (io_unit,'(a18,1X,i2.2,"/",i2.2,"/",i4.4,1X,a2,1X,i2.2,2(":",i2.2))') &
                    "Started running on", date_values(3:1:-1), "at", date_values(5:7)
-        write (io_unit,'(1X,"Calculation UUID:",1X,a36,".")') GLOBAL_UUID
-
-        write (io_unit,'(1X,"Preprocessor settings:")')
-        ! Sadly preprocessor does not retokenize the output, so have to do this by hand...
-#ifdef DISABLE_HDF5
-        write (io_unit,'(5X,"DISABLE_HDF5 defined.  HDF5 disabled.")')
-#else
-        write (io_unit,'(5X,"DISABLE_HDF5 not defined.  HDF5 enabled.")')
-#endif
-#ifdef DISABLE_LANCZOS
-        write (io_unit,'(5X,"DISABLE_LANCZOS defined.  Lanczos disabled.")')
-#else
-        write (io_unit,'(5X,"DISABLE_LANCZOS not defined.  Lanczos enabled.")')
-#endif
-#ifdef DISABLE_UUID
-        write (io_unit,'(5X,"DISABLE_UUID defined.  UUID disabled.")')
-#else
-        write (io_unit,'(5X,"DISABLE_UUID not defined.  UUID enabled.")')
-#endif
-#ifdef PARALLEL
-        write (io_unit,'(5X,"PARALLEL defined.  MPI parallelization enabled.")')
-#else
-        write (io_unit,'(5X,"PARALLEL not defined.  MPI parallelization disabled.")')
-#endif
-#ifdef DISABLE_SCALAPACK
-        write (io_unit,'(5X,"DISABLE_SCALAPACK defined.  ScaLAPACK disabled.")')
-#else
-        write (io_unit,'(5X,"DISABLE_SCALAPACK not defined.  ScaLAPACK enabled.")')
-#endif
-#ifdef SINGLE_PRECISION
-        write (io_unit,'(5X,"SINGLE_PRECISION defined.  Single precision used where relevant.")')
-#else
-        write (io_unit,'(5X,"SINGLE_PRECISION not defined.  Double precision used throughout.")')
-#endif
-#ifdef USE_POPCNT
-        write (io_unit,'(5X,"USE_POPCNT defined.  Fortran 2003 POPCNT procedure used.")')
-#else
-        write (io_unit,'(5X,"USE_POPCNT not defined.  Internal POPCNT procedure used.")')
-#endif
-        write (io_unit,'(5X,"DET_SIZE = ",i2,".")') bit_size(0_i0)
-        write (io_unit,'(5X,"POP_SIZE = ",i2,".")') bit_size(0_int_p)
+        write (io_unit,'("Calculation UUID:",1X,a36)') new_line('a')//GLOBAL_UUID
```

Comment:
Why the change for the output format of the UUID line with a new line? Also (for consistency) prefer using / in for format string instead of new_line.

Reply:
I thought it would fit better with the rest of the printing for that section. But on second thought it looks uglier, so I'll revert it.